db/schema.rb

```diff
@@ -76,6 +76,17 @@
 
   add_index "linksets", ["rubygem_id"], name: "index_linksets_on_rubygem_id", using: :btree
 
+  create_table "log_tickets", force: :cascade do |t|
+    t.string   "key"
+    t.string   "directory"
+    t.integer  "backend",    default: 0
```

Comment:
One solution I could do is, instead of having a `backend` flag on the log file level, I could add a ENV var that would either pull the log from s3 or local fs.
Only draw back about this, is that if we ever wanna fix one file that went wrong, and we wanted to manually drop that file in production some where and have the process working on it, we would have to hack something on top of that.

Also, the extra tests are like 3 tests, 12 lines of code. As the facade itself(RubygemsFS) is already been tested as a unit.

I think having a backend per log, will give us more flexibility and it actually hasn't cost much in terms of code, so thats why I am pushing this approach.

Also, this allow us to process files from other backends in the future, lets say we start pushing logs to kafka, and we want to process them. I know this sounds a bit like crazy talk now, but for the current use case we have, having two backends sounded like a good abstraction area to me.